store/sqlstore/channel_store.go

```diff
@@ -3751,3 +3751,47 @@ func (s SqlChannelStore) UpdateSidebarChannelCategoryOnMove(channel *model.Chann
 	}
 	return nil
 }
+
+// DeleteSidebarCategory removes a custom category and moves any channels into it into the Channels and Direct Messages
+// categories respectively. Assumes that the provided user ID and team ID match the given category ID.
+func (s SqlChannelStore) DeleteSidebarCategory(categoryId string) *model.AppError {
+	transaction, err := s.GetMaster().Begin()
+	if err != nil {
+		return model.NewAppError("SqlChannelStore.DeleteSidebarCategory", "store.sql_channel.sidebar_categories.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	defer finalizeTransaction(transaction)
+
+	// Ensure that we're deleting a custom category
```

Comment:
I think we shouldn't check this at store later, it would be much convenient in the App layer IMHO.

Reply:
this is done to avoid unnecessary DB call on the App side, and to avoid creating another category getter

Reply:
I see, I'm not above 0/5 so it's ok :)

Reply:
I'm 0/5 about where it goes as well. I think having it in a different layer makes sense in some ways since that's where we enforce most business rules, but having it here can also make sense since it's about ensuring data integrity by preventing us from accidentally deleting a non-custom category. I'm kind of leaning towards the database integrity thing here, but I'm definitely open to change it.

Reply:
I see, there are similar occurrences in the store layer so this won't stand out. It's up to you, I just wanted to point it out :)